contracts/tokenbridge/ethereum/L1AtomicTokenBridgeCreator.sol

```diff
@@ -233,17 +248,22 @@ contract L1AtomicTokenBridgeCreator is Initializable, OwnableUpgradeable {
         return getCanonicalL1RouterAddress(inbox);
     }
 
-    function _deployL1Contracts(address inbox, address owner, bool isUsingFeeToken)
+    function _deployL1Contracts(address inbox, address rollupOwner, bool isUsingFeeToken)
         internal
         returns (
             address router,
             address standardGateway,
             address customGateway,
-            address wethGateway
+            address wethGateway,
+            address proxyAdmin
         )
     {
-        address proxyAdmin =
-            address(new ProxyAdmin{ salt: _getL1Salt(OrbitSalts.L1_PROXY_ADMIN, inbox) }());
+        // get existing proxy admin and upgrade executor
+        proxyAdmin = IInbox_ProxyAdmin(inbox).getProxyAdmin();
+        if (proxyAdmin == address(0)) {
+            revert L1AtomicTokenBridgeCreator_ProxyAdminNotFound();
+        }
+        address upgradeExecutor = ProxyAdmin(proxyAdmin).owner();
```

Comment:
Maybe a safety check just to see if this address looks like an UpgradeExecutor - maybe has the correct role properties like EXECUTOR_ROLE. Just to stop an accidental mistake

Reply:
Good point, there were actually some inconsistencies because latest rollup ownership design was not fully reflected. Latest design is this - rollup and rollup proxy admin are owned by L1 upgrade executor on which the rollupOwner has EXECUTOR role (initially rollup.owner() was rollupOwner, not upgradeExecutor).  

This required a bit of refactoring. Idea is this:
- on L1, use existing rollup's proxyAdmin and make existing L1 upgrade executor the owner of ownable stuff
- on L2, create new L2 upgrade executor. Aliased L1 upg executor and rollupOwner are are given EXECUTOR role

One inconvenience is that we can't dynamically figure out who is the rollupOwner for given rollup. That's because `rollupOwner` just has executor role on L1 upgrade executor, and we're not using enumberable access control. For that reason I updated `createTokenBridge` entrypoint to additionally receive rollupOwner address. It will then be verified by checking it has correct role on the upgradeExecutor which we get as rollup.owner().

Implementation here - https://github.com/OffchainLabs/token-bridge-contracts/pull/38/commits/2ecec9a9ddf14874a39b7d827726ef32e4049251

Reply:
Makes sense